ansible/ansible-ai-connect-service
# AAP-30986: Support deletion of WCA API_KEY from Admin Portal (Frontend)

Jira Issue: <https://issues.redhat.com/browse/AAP-30986>


## Description
Please see the Jira issue for details. Note that this issue splitted into two: frontend & backend. This PR includes the FE changes work.

## Testing
Follow the following steps. Do not commit anything by mistake as you will need to de-comment some places in the code.
### Steps to test
1. Go to ansible_ai_connect_admin_portal/src/ModelSettingsOverview.tsx:288
2.  Comment the related part (the delete button)
3. Run `npm start` Open the link in browser tab. This is the mocked mode.
4. Test out the deletion. You might need to play with the mock API here (ansible_ai_connect_admin_portal/config/webpack.mock.api.keys.ts:66) to simulate an error.
5. Revert your changes and dont commit/push them by mistake.
6. Run the `npm run test` command to make sure all tests pass.

### Scenarios tested
* Deletion with success
* Deletion with failure

## Production deployment

- [x] This code change requires the following considerations before going to production: The backend part of this code should be completed and done. See AAP-32688.
